### ISSUE TITLE
Using <fieldset> without a $Legend fails WCAG 2.0 Accessibility

### ISSUE THREAD

**Issue by aNickzz** [silverstripe/silverstripe-framework#8227]
## Affected Version

All

## Description

This is really a copy of https://github.com/silverstripe/silverstripe-userforms/issues/515. To quote the article linked in that issue:

> The first element inside the fieldset must be a legend element, which provides a label or description for the group.

This has also come up before in #5733, but that issue has gone stale.

A resolution similar to that used in Userforms would be idea, however I'd also request that both the `<fieldset>` and the `<div>` both share the same class to make styling easier.

```
<% if $Legend %>
    <fieldset class="fields">
        <legend>$Legend</legend>
<% else %>
    <div class="fields">
<% end_if %>

        <% loop $Fields %>
            $FieldHolder
        <% end_loop %>

<% if $Legend %>
    </fieldset>
<% else %>
    </div>
<% end_if %>
```

I don't know the extend to which this will affect existing styles / javscript.

**Comment by dhensby** [comment]
Yeah, the problem is that this could be a pretty big deal for people who have CSS that rely on it. We can only really change this in master (and we should) - however, I am going to close this as a duplicate of #5733 just because it's not moved for a while doesn't mean it's not a good canonical source for the issue
